pkg/ingresscache/safetrie_test.go

```diff
@@ -0,0 +1,449 @@
+/*
+Copyright 2025 Iguazio Systems Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License") with
+an addition restriction as set forth herein. You may not use this
+file except in compliance with the License. You may obtain a copy of
+the License at http://www.apache.org/licenses/LICENSE-2.0.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing
+permissions and limitations under the License.
+
+In addition, you may not use the software for any purposes that are
+illegal under applicable law, and the grant of the foregoing license
+under the Apache 2.0 license is conditioned upon your compliance with
+such restriction.
+*/
+package ingresscache
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type SafeTrieTestSuite struct {
+	suite.Suite
+}
+
+type safeTrieFunctionArgs struct {
+	path     string
+	function string
+}
+
+func (suite *SafeTrieTestSuite) TestPathTreeSet() {
+	suite.T().Parallel()
+	for _, testCase := range []struct {
+		name           string
+		args           []safeTrieFunctionArgs
+		expectedResult map[string][]string
+		shouldFail     bool
+		errorMessage   string
+	}{
+		{
+			name: "simple set",
+			args: []safeTrieFunctionArgs{
+				{
+					path:     "/path/to/function",
+					function: "test-function",
+				},
+			},
+			expectedResult: map[string][]string{"/path/to/function": {"test-function"}},
+		}, {
+			name: "idempotent test",
+			args: []safeTrieFunctionArgs{
+				{
+					path:     "/path/to/function",
+					function: "test-function",
+				}, {
+					path:     "/path/to/function",
+					function: "test-function",
+				},
+			},
+			expectedResult: map[string][]string{"/path/to/function": {"test-function"}},
+		}, {
+			name: "set twice the same path with a different function",
+			args: []safeTrieFunctionArgs{
+				{
+					path:     "/path/to/function",
+					function: "test-function",
+				}, {
+					path:     "/path/to/function",
+					function: "test-function2",
+				},
+			},
+			expectedResult: map[string][]string{"/path/to/function": {"test-function", "test-function2"}},
+		}, {
+			name: "set nested paths and different functions",
+			args: []safeTrieFunctionArgs{
+				{
+					path:     "/path/to/function",
+					function: "test-function",
+				}, {
+					path:     "/path/to/function/nested",
+					function: "test-function2",
+				},
+			},
+			expectedResult: map[string][]string{
+				"/path/to/function":        {"test-function"},
+				"/path/to/function/nested": {"test-function2"},
+			},
+		}, {
+			name: "set different paths and different functions",
+			args: []safeTrieFunctionArgs{
+				{
+					path:     "/path/to/function",
+					function: "test-function",
+				}, {
+					path:     "/another/path/to/function/",
+					function: "test-function2",
+				},
```

Comment:
what will happen in this case? and how it aligns with ingress behaviour in this case
```suggestion
		}, {
			name: "set different paths and different functions",
			args: []safeTrieFunctionArgs{
				{
					path:     "/path/to/function",
					function: "test-function",
				}, {
					path:     "/path/to/function/",
					function: "test-function2",
				},
```

Reply:
There are two separate points here:

It's not possible to define two ingresses that differ only by a trailing /. For example, if there's already an ingress with the path /path/to/function, trying to add another one with /path/to/function/ will fail during the [Dashboard's ingress validation](https://github.com/nuclio/nuclio/blob/995fa4289b1551b685cd44d44bfa521709496b21/pkg/platform/kube/platform.go#L2192).
This happens because the validation logic normalizes the URL before checking for duplicates — as seen [here](https://github.com/nuclio/nuclio/blob/826dfafdf71b2214c02f45de8c14af083d655432/pkg/common/url.go#L106).

The goal of this test is to verify behavior with two distinct paths, not nested ones — so I’d prefer to keep it as is.